src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -79,16 +103,18 @@ void Ekf::controlGpsFusion()
 				if (continuing_conditions_passing
 				    || !isOtherSourceOfHorizontalAidingThan(_control_status.flags.gps)) {
 
-					if (_params.gnss_ctrl & GnssCtrl::VEL) {
-						fuseGpsVel();
-					}
+					_aid_src_gnss_vel.fusion_enabled[0] = (_params.gnss_ctrl & GnssCtrl::VEL) && _control_status.flags.gps;
```

Comment:
we can only be here when `_control_status.flags.gps` is `true`

Reply:
I was getting ready to open this up for gps_hgt as well (until we talked about it the other day). 

After that conversation I have a big set of changes splitting gps_hgt, ev_hgt entirely. Let me either finish that up here or see if it can be brought in incrementally on the side.

Reply:
Split in https://github.com/PX4/PX4-Autopilot/pull/20397.